### PR TITLE
🐛 Skip publishing the `RemediationRestricted` event when there are no unhealthy target

### PR DESCRIPTION
**Pull request by Levi080513**
**What this PR does / why we need it**:

When there are no unhealthy target, skip publishing the `RemediationRestricted` event to avoid misleading.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes # https://github.com/kubernetes-sigs/cluster-api/issues/10585

/area machinehealthcheck

### CODE REVIEW

**Comment by enxebre** [comment]
Thanks!
/ok-to-test
/lgtm

File: internal/controllers/machinehealthcheck/machinehealthcheck_controller.go [hunk 1]
```diff
@@ -279,9 +279,15 @@ func (r *Reconciler) reconcile(ctx context.Context, logger logr.Logger, cluster
 			Message:  message,
 		})
 
+		eventType := corev1.EventTypeWarning
```
Comment (sbueringer) [inline internal/controllers/machinehealthcheck/machinehealthcheck_controller.go:282]: Should we even publish an event in that case?

Comment (Levi080513) [reply internal/controllers/machinehealthcheck/machinehealthcheck_controller.go:282]: In this case, the event doesn't seem to make any sense, I'll fix it.

**Comment by sbueringer** [comment]
Fine for me

**Pull request merged**